### DISCUSSION TITLE
Should zopen be enhanced to detect vulnerabilities for installed software?

### DISCUSSION THREAD

**Discussion by IgorTodorovskiIBM** [zopencommunity/meta#214] [Q&A]
We could invent a new audit function similar to npm audit - https://docs.npmjs.com/auditing-package-dependencies-for-security-vulnerabilities

`zopen audit`

Which could use the https://nvd.nist.gov/ APIs to check for vulnerabilities for a given product/version and recommend upgrading (if a newer version is present).

**Comment by MikeFultonDev** [comment]
We haven't done anything here yet afaik, but this is a good idea @IgorTodorovskiIBM

**Reply by IgorTodorovskiIBM** [reply]
This may be a better option: https://google.github.io/osv.dev/. 

The API can be called via curl and therefore integrated in zopen pretty easily:
```
curl -d \
  '{"package": {"name": "curl"}, "version": "7.88.1-10"}' \
  "https://api.osv.dev/v1/query"
```
 
There's also an OSV-scanner written in Go: https://github.com/google/osv-scanner

**Reply by v1gnesh** [reply]
Here's an API for a superset of this, also run by Google - https://deps.dev/

**Reply by IgorTodorovskiIBM** [reply]
Thanks, the https://deps.dev/ API seems to be more powerful. It looks like it doesn't seem to support C/C++ but it does support Go packages.

**Reply by v1gnesh** [reply]
Ah, yes. The API docs delegates advisory lookup to OSV but the advisory # is needed.
